python/semantic_kernel/connectors/openapi_plugin/openapi_manager.py

```diff
@@ -46,12 +47,18 @@ def create_functions_from_openapi(
         list[KernelFunctionFromMethod]: the operations as functions
     """
     parser = OpenApiParser()
-    parsed_doc = parser.parse(openapi_document_path)
+    if (parsed_doc := parser.parse(openapi_document_path)) is None:
+        raise FunctionExecutionException(f"Error parsing OpenAPI document: {openapi_document_path}")
     operations = parser.create_rest_api_operations(parsed_doc, execution_settings=execution_settings)
 
     auth_callback = None
     if execution_settings and execution_settings.auth_callback:
         auth_callback = execution_settings.auth_callback
+
+    # For type chceker, ensure that parsed_doc is a Mapping[str, str]
+    if isinstance(parsed_doc, Mapping):
+        parsed_doc = {k: str(v) for k, v in parsed_doc.items()}
```

Comment:
do we need this loop, couldn't we check the type of `v` in the parser?

Reply:
This was needed to satisfy a mypy type check, but strangely, when I comment out the loop, mypy isn't throwing an error. Removed it.